blueprint/tests/blueprint_EcmascriptEngineTests.h

```diff
@@ -0,0 +1,234 @@
+        expect (obj->getProperty ("PLUGIN_VERSION").toString() == "1.3.5");
+        expect (obj->getProperty ("PLUGIN_NAME").toString() == "Temper");
+    }
+};
+
+//==============================================================================
+class EcmascriptEngineInvokeTest final : public BlueprintUnitTest
+{
+public:
+    EcmascriptEngineInvokeTest() : BlueprintUnitTest ("Function Invocation Interfaces") {}
+
+    void runTest() override
+    {
+        beginTest ("Invoking Builtins");
+        expect (engine.invoke ("String.fromCharCode", 42).toString() == "*");
+        expect (static_cast<int> (engine.invoke ("parseInt", 1.00031439, 10)) == 1);
+        expect (static_cast<int> (engine.invoke ("Math.abs", -1)) == 1);
+
+        beginTest ("Invoking Native Methods");
+        engine.registerNativeProperty ("BlueprintNative", juce::JSON::parse ("{}"));
+
+        engine.registerNativeMethod ("BlueprintNative", "squareIt",
+        [&](const juce::var::NativeFunctionArgs& args)
+        {
+            return juce::square (static_cast<int> (args.arguments[0]));
+        });
+
+        expect (static_cast<int> (engine.invoke ("BlueprintNative.squareIt", 2)) == 4);
+        expect (static_cast<int> (engine.invoke ("BlueprintNative.squareIt", 3)) == 9);
+    }
+};
+
+//==============================================================================
+class EcmascriptEngineErrorHandlerTest final : public BlueprintUnitTest
+{
+public:
+    EcmascriptEngineErrorHandlerTest() : BlueprintUnitTest ("Error Handler Interfaces") {}
+
+    void runTest() override
+    {
+        auto noopFunc = [&] (const juce::var::NativeFunctionArgs&)
+        {
+            return juce::var::undefined();
+        };
+
+        auto testAndExpectThrow = [&] (auto&& func)
+        {
+            bool didThrow = false;
+
+            try
+            {
+                func();
+            }
+            catch (...)
+            {
+                didThrow = true;
+            }
+
+            expect (didThrow);
+        };
+
+        beginTest ("Parse Errors");
+
+        testAndExpectThrow ([&]() { evaluate ("1 + 2 + "); });
+        testAndExpectThrow ([&]() { invoke ("Blueprint.1+", 1); });
+        testAndExpectThrow ([&]() { engine.registerNativeMethod ("global[1 + 2 +]", "Noop", noopFunc); });
+        testAndExpectThrow ([&]() { engine.registerNativeProperty ("global[1 + 2 +]", "Noop", 42); });
+
+        beginTest ("Runtime Errors");
+
+        testAndExpectThrow ([&]() { evaluate ("doesNotExist();"); });
+        testAndExpectThrow ([&]() { invoke ("Blueprint[doesNotExist()]", 1); });
+        testAndExpectThrow ([&]() { engine.registerNativeMethod ("global[doesNotExist()]", "Noop", noopFunc); });
+        testAndExpectThrow ([&]() { engine.registerNativeProperty ("global[doesNotExist()]", "Noop", 42); });
+    }
+};
+
+//==============================================================================
+struct SourceFileTest : blueprint::BlueprintUnitTest
```

Comment:
@nick-thompson Check this puppy out...

It works in MSVC rather well. I suggest giving this PR a whirl locally, and have a look at the TestRunner now (rather beefed it up).

Reply:
> Maybe we could get this merged first, then do the cmake transition, then do a test runner addition for the gain plugin all separately?

Yes! Sounds great